puppet/manifests/default.pp

```diff
@@ -30,8 +79,27 @@
   }
 }
 
+# This needs to be on exactly two nodes.
+class gluster_cluster {
+  include glusterfs_common
+
+  gluster_peer { ['controller.seed-stack.local', 'worker.seed-stack.local']: }
```

Comment:
An autorequire is only used for ordering, not actual dependencies. Anything that's autorequired but not part of the catalog is ignored.

The idea is that you can autorequire various things for different operating systems or whatever and the irrelevant things will be ignored.